clients/branch.go

```diff
@@ -28,13 +28,13 @@ type BranchProtectionRule struct {
 	AllowForcePushes           *bool
 	RequireLinearHistory       *bool
 	EnforceAdmins              *bool
-	RequiredStatusChecks       StatusChecksRule
+	RequiredStatusChecks       *StatusChecksRule
```

Comment:
The data structure should be explicit about what information it captures. Using a `nil` `RequiredStatusChecks` struct to check for *status checks are not required* is confusing and is against the convention we are trying to build in #1242. Instead, how about something like:

```
type StatusChecksRule struct {
  RequiresStatusChecks *bool
  Strict/UpToDate *bool
  Contexts []string
}
``` 

Basically, we don't use pointers to `struct` and instead use pointers to primitive types.

Reply:
I did exactly that: use pointer to struct :-)
I'll update to using pointers to primitives. Thanks!